doc/document/PortingToFreeRTOSAWS.md

```diff
@@ -0,0 +1,298 @@
+    PRIVATE
+        "${CMAKE_CURRENT_LIST_DIR}/library/modules/bg96/cellular_bg96.c"
+        "${CMAKE_CURRENT_LIST_DIR}/library/modules/bg96/cellular_bg96_api.c"
+        "${CMAKE_CURRENT_LIST_DIR}/library/modules/bg96/cellular_bg96_urc_handler.c"
+        "${CMAKE_CURRENT_LIST_DIR}/library/modules/bg96/cellular_bg96_wrapper.c"
+)
+
+afr_module_dependencies(
+    cellular_module
+    PRIVATE
+        AFR::cellular
+        AFR::cellular_common
+)
+```
+
+### The sample code for board CMakeLists.txt to include cellular library
+
+The “AFR::cellular::mcu_port” module is defined in this sample code.
+
+```cmake
+# FreeRTOS Cellular Library comm interface
+afr_mcu_port(cellular)
+target_sources(
+    AFR::cellular::mcu_port
+    INTERFACE "${afr_ports_dir}/comm_if/comm_if_windows.c"
+)
+
+# FreeRTOS Cellular Library
+include("${afr_ports_dir}/cellular/CMakeLists.txt")
+
+....
+
+# Do not add demos or tests if they're turned off.
+if(AFR_ENABLE_DEMOS OR AFR_ENABLE_TESTS)
+    add_executable(
+        ${exe_target}
+        "${board_dir}/application_code/main.c"
+        "${board_demos_dir}/application_code/aws_demo_logging.c"
+        "${board_demos_dir}/application_code/aws_demo_logging.h"
+        "${board_demos_dir}/application_code/aws_entropy_hardware_poll.c"
+        "${board_demos_dir}/application_code/aws_run-time-stats-utils.c"
+        "${board_demos_dir}/application_code/cellular_setup.c" 
+        # Add cellular_setup to enable cellular network
+    )
+    target_include_directories(
+        ${exe_target}
+        PRIVATE
+            "${board_demos_dir}/application_code"
+    )
+    target_link_libraries(
+        ${exe_target}
+        PRIVATE
+            AFR::freertos_plus_tcp
+            AFR::utils
+            AFR::dev_mode_key_provisioning
+            AFR::cellular # Add AFR::cellular dependency
+    )
+endif()
+
+```
+
+### The sample code for board CMakeLists.txt to include secure sockets cellular
+
+The “AFR::secure_sockets::mcu_port” is defined in this sample code.
+
+```cmake
+
+# Secure sockets
+afr_mcu_port(secure_sockets)
+target_sources(
+    AFR::secure_sockets::mcu_port
+    INTERFACE
+        "${afr_ports_dir}/cellular/secure_sockets/iot_secure_sockets.c"
+)
+
+target_include_directories(
+    AFR::secure_sockets::mcu_port
+    INTERFACE
+        "${afr_ports_dir}/cellular/configs"
+        "${afr_ports_dir}/cellular/library/include"
+)
+
+target_link_libraries(
+    AFR::secure_sockets::mcu_port
+    INTERFACE 
+        AFR::platform
+        AFR::tls
+        AFR::crypto
+        AFR::pkcs11
+)
+```
+
+### The sample code for the main.c file
+
+This is the sample code for windows simulator aws_demos. FreeRTOS Cellular Library will create one receive thread in Cellular_Init function. Developer has to make sure to call setupCellular in context which can create thread.
+
+```c
+void vApplicationIPNetworkEventHook( eIPCallbackEvent_t eNetworkEvent )
+{
+    ... 
+    /* If the network has just come up...*/

```

Comment:
```suggestion
    /* If the network has just come up*/

```

Reply:
This is the original code snippet. It is not updated in this guide.